### DISCUSSION TITLE
How do I modify a response from interaction?

### DISCUSSION THREAD

**Discussion by marss72** [discord-net/Discord.Net#2992] [Q&A] [answered]
Please take a look at this example.

``` cs
[SlashCommand("example", "Command description.")]
public async Task CommandAsync()
{
    await RespondAsync("Please click this button.",
        components: new ComponentBuilder()
            .WithSelectMenu(
                new SelectMenuBuilder()
                    .WithCustomId("click-submit")
                    .WithOptions([new("First option", "first"), new("And second option", "second")])
            )
            .Build(),
        ephemeral: true
    );
}

[ComponentInteraction("click-submit")]
public async Task SubmitReport(string[] selected)
{
    // ...
}
```

How do I modify the message sent from CommandAsync when the button is clicked? Is that even possible? I tried using `ModifyOriginalResponseAsync`, but that always throws `Discord.Net.HttpException: The server responded with error 10015: Unknown Webhook`.

In short, I need to modify the message sent from `CommandAsync` in `SubmitReport` to something else, including removing all the components.

What am I doing wrong? Thanks.

**Comment by Misha-133** [comment, marked as answer]
in the `SubmitReport` method do something like 
```cs
var interaction = (IComponentInteraction)Context.Interaction;
await interaction.UpdateAsync(x => x.XXXX);
```

Also [these](https://docs.discordnet.dev/faq/int_framework/respondings-schemes.html#responding-to-a-component-interaction) might come helpful

**Reply by marss72** [reply]
Thanks a lot!

**Reply by marss72** [reply]
@Misha-133, is it also possible to delete the message? Don't see any method for that.

**Reply by Misha-133** [reply]
Not on original response, you have to use `DeferAsync`/`UpdateAsync` first
![image](https://github.com/user-attachments/assets/ccb58ee0-f444-42ca-b8e5-e36374add389)
